eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Procedures.hs

```diff
@@ -260,15 +261,24 @@ data GovernanceProcedures era = GovernanceProcedures
   }
   deriving (Eq, Generic)
 
+-- | Attaches indices to a sequence of proposal procedures. The indices grow
+-- from left to right.
+indexedGovProps ::
+  Seq (ProposalProcedure era) ->
+  Seq (GovernanceActionIx, ProposalProcedure era)
+indexedGovProps = enumerateProps 0
+  where
+    enumerateProps _ Empty = Empty
+    enumerateProps n (x :<| xs) = (n, x) :<| enumerateProps (succ n) xs
+
 instance EraPParams era => NoThunks (GovernanceProcedures era)
 
 instance EraPParams era => NFData (GovernanceProcedures era)
 
 deriving instance EraPParams era => Show (GovernanceProcedures era)
 
 data ProposalProcedure era = ProposalProcedure
-  { pProcDeposit :: !Coin
```

Comment:
Basically so that we don't have to store the deposit twice. But I do agree that it'd be nice to have it part of the transaction. But then we'd have two different types for the transaction & what we store.

Reply:
I don't quite get it. We don't store the `ProposalProcedure` in memory. It is an ephemeral data type that is alive as long as the transaction is being validated.

It would be placed on chain, but that's a different story.

Reply:
I restored the `pProcDeposit` field